### PR TITLE
Scripts/Spells: Why double implementation Nurturing Instinct

### PR DESCRIPTION
**Pull request by HoustonCore**
## CHANGES PROPOSED:
-  Why double implementation of the talent Nurturing Instinct in SpellAuraEffect and spell_druid????

## TESTS PERFORMED:
- Build and test in game

## HOW TO TEST THE CHANGES:

- Create druid
- Learn talent Nurturing Instinct
- Use heal in form cat and see

## Target branch(es):
- [x] Master

 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

File: src/server/scripts/Spells/spell_druid.cpp [hunk 1]
```diff
@@ -100,16 +101,17 @@ class spell_dru_nurturing_instinct : public SpellScriptLoader
         {
             PrepareAuraScript(spell_dru_nurturing_instinct_AuraScript);
 
-            void AfterApply(AuraEffect const*  /*aurEff*/, AuraEffectHandleModes /*mode*/)
+            void AfterApply(AuraEffect const* aurEff, AuraEffectHandleModes /*mode*/)
             {
-                if (Player* player = GetTarget()->ToPlayer())
-                    player->addSpell(GetSpellInfo()->GetRank() == 1 ? SPELL_DRUID_NURTURING_INSTINCT_R1 : SPELL_DRUID_NURTURING_INSTINCT_R2, SPEC_MASK_ALL, false, true);
+                Unit* target = GetTarget();
```
[the remark this answers is not in the document]
Comment (Viste) [reply src/server/scripts/Spells/spell_druid.cpp:106]: no its not safe ) nice catch

[the remark this answers is not in the document]
Comment (Goatform) [reply src/server/scripts/Spells/spell_druid.cpp:106]: Tnx for confirming @Viste 

Well, then @HoustonCore just add:

```
Unit* target = GetTarget();
if (!target)
    return;
```

File: src/server/scripts/Spells/spell_druid.cpp [hunk 2]
```diff
@@ -101,17 +101,16 @@ class spell_dru_nurturing_instinct : public SpellScriptLoader
...
-                target->CastSpell(target, spellId, aurEff);
+                if (Player* player = GetTarget()->ToPlayer())
+                    player->addSpell(GetSpellInfo()->GetRank() == 1 ? SPELL_DRUID_NURTURING_INSTINCT_R1 : SPELL_DRUID_NURTURING_INSTINCT_R2, SPEC_MASK_ALL, false, true);
             }
 
             void AfterRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
             {
-                uint32 spellId = GetSpellInfo()->GetRank() == 1 ? SPELL_DRUID_NURTURING_INSTINCT_R1 : SPELL_DRUID_NURTURING_INSTINCT_R2;
-                GetTarget()->RemoveAurasDueToSpell(spellId);
+                if (Player* player = GetTarget()->ToPlayer())
```
Comment (Goatform) [inline src/server/scripts/Spells/spell_druid.cpp]: It still only checks for ToPlayer() ... how about GetTarget() ?

**Comment by Racknar0** [comment]
Tested!

before RP

image

after the PR

image

Conclusion:

The druid after the PR continues healing the same by applying the talent, tested with the same druid the same class and the same amount of agility

**Pull request closed without merging**